File: other/backlight.rs
//Soure code for my backlight control program, rewritten from previous bash version
use std::{
	fs::{self, File},
	io::Write,
};

fn main() {
	match change() {
		Ok(_) => (),
		Err(_) => {
			eprintln!("Please do sudo chown root:root ./backlight && sudo chmod +s ./backlight")
		}
	}
}
fn change() -> Result<(), Box<dyn std::error::Error>> {
	let args: Vec<String> = std::env::args().collect();
	let path = fs::read_dir("/sys/class/backlight")?.next().unwrap()?.path();
	let now_path = path.join("brightness");
	let mut now: u32 = fs::read_to_string(&now_path)?.trim_end().parse()?;
	let max: u32 = fs::read_to_string(path.join("max_brightness"))?.trim_end().parse()?;
	if args.len() > 1 {
		match args[1].as_str() {
			"↓" => now = if now < 2 { 0 } else { now * 2 / 3 - 1 },
			"↑" => {
				now = now * 3 / 2 + 3;
				if now > max {
					now = max
				}
			}
			"-" => {
				let new = args[2].as_str().parse::<u32>()? * max / 100;
				now = if now < new { 0 } else { now - new };
			}
			"+" => {
				let new = args[2].as_str().parse::<u32>()? * max / 100;
				now = if now + new > max { max } else { now + new };
			}
			"=" => {
				now = {
					let new = args[2].as_str().parse::<u32>()? * max / 100;
					if new > max {
						max
					} else {
						new
					}
				};
			}
			_ => (),
		}
		File::create(now_path)?.write_all(now.to_string().as_bytes())?;
	}
	println!("{}", 100 * now / max);
	Ok(())
}

File: other/backlight.c
#include <dirent.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Soure code for my backlight control program, rewritten from rust 'coz size too big
int main(int argc, char* argv[]) {
	DIR* d             = opendir("/sys/class/backlight/");
	struct dirent* dir = readdir(d);
	while (dir && dir->d_type != DT_LNK) dir = readdir(d);
	closedir(d);
	if (!dir) return EXIT_FAILURE;
	char str[55];
	int dirEnd = 20;
	strcpy(str, "/sys/class/backlight/");
	strcat(str, dir->d_name);
	while (str[dirEnd]) dirEnd++;
	str[dirEnd++] = '/';

	str[dirEnd]   = 0;
	strcat(str + dirEnd, "max_brightness");
	FILE* file = fopen(str, "r");
	if (!file) return EXIT_FAILURE;
	int max = 0;
	for (char c; (c = fgetc(file)) > 47;) max = (max * 10) + c - 48;
	fclose(file);

	str[dirEnd] = 0;
	strcat(str + dirEnd, "brightness");
	if (!(file = fopen(str, "r+"))) {
		printf("Please do sudo chown root:root ./backlight && sudo chmod +s ./backlight");
		return EXIT_FAILURE;
	}
	int now = 0;
	for (char c; (c = fgetc(file)) > 47;) now = (now * 10) + c - 48;
	if (argc > 1) {
		if (argv[1][0] == 'v') {
			if (now < 3) now = 0;
			else now = now * 2 / 3 - 1;
		} else if (argv[1][0] == '^') now = now * 3 / 2 + 3;
		else if (argc < 3) {
			fclose(file);
			return EXIT_FAILURE;
		} else {
			int val = atoi(argv[2]) * max / 100;
			if (argv[1][0] == '-') now = now >= val ? now - val : 0;
			else if (argv[1][0] == '+') now += val;
			else if (argv[1][0] == '=') now = val;
			else {
				printf("usage: backlight [v^][[+-=] int]\n");
				return EXIT_FAILURE;
			}
		}
		if (now > max) now = max;

		sprintf(str, "%d", now);
		for (int i = 0; str[i]; i++) fputc(str[i], file);
	}
	fclose(file);
	printf("%d\n", now * 100 / max);

	return EXIT_SUCCESS;
}
